### ISSUE TITLE
None of the allowed keys is able to decrypt the content

### ISSUE THREAD

**Issue by tgrosinger** [iNPUTmice/Conversations#2073]
#### General information
- **Version:** eg 1.14.6
- **Device:** eg Google Nexus 6P
- **Android Version:** Android 6.0 Stock
- **Server name:** self hosted
- **Server software:** ejabberd 16.04 or prosody 0.10 (if known)
- **Installed server modules:** adhoc, admin_adhoc, auth_internal_plain, c2s, carbons, csi, dialback, disco, filter_chatstates, http, iq, mam, mam_muc, message, offline, pep, ping, presence, private, register, roster, s2s, saslauth, smacks, storage_sql, throttle_presence, time, tls, uptime, vcard, version
- **Conversations source:** F-Droid
#### Steps to reproduce
1. Install conversations and openkeychain. Add your private key to openkeychain
2. Import friends public keys on openkeychain. Note, they are all signed and trusted.
3. Add friends to roster in conversations.
4. Have friendly conversation using OpenPGP encryption
5. After an unknown cause, one friend (this has happened sporadically with all three people I am testing with) loses the ability to decrypt new message.

When this happens conversations continually prompts to decrypt OpenPGP messages, and when clicking this the OpenKeychain window opens asking me to "select the allowed keys". After selecting the right key and clicking save the window disappears and nothing happens.

All clients I am testing with are using the same server, same model phone, same version conversations/openkeychain.

The only fix found so far is to clear all app data for conversations and openkeychain.

One last piece of the puzzle. I have another client installed with access to the same set of public and private keys. It too was working, but when my phone stopped working, the pgp encrypted messages stopped showing up there. They appear as completely blank messages instead of the expected "I sent you an OpenPGP encrypted message but your client doesn't seem to support that."

**Comment by licaon-kter** [comment]
Logs?

**Comment by akater** [comment]
I have the same issue: Unable to decrypt › OpenKeychain › Proper key is already selected › I press Save › No effect. The message was sent from Gajim. Logs won't help^ it's just date, JID and ciphertext there.

I'm 99% sure the encryption key was correct but can't make sure: messages (i.e., the output of `message.getBody()` in https://github.com/siacs/Conversations/blob/9d0a33337218fbbd8852239fad374f495b4a4243/src/main/java/eu/siacs/conversations/services/ExportLogsService.java#L105 ) from log file are not recognised by `gpg` as valid OpenPGP data.

**Comment by akater** [comment]
In case maintainer(s) need someone for field tests with PGP, I'll be at your service from 2016-10-16 20:00 UTC.

**Comment by iNPUTmice** [comment]
Maybe @dschuermann can shed on light on this. I'm calling the OpenKeychain API with a very simple `ACTION_DECRYPT_VERIFY`. No further parameters. See [this code](https://github.com/siacs/Conversations/blob/master/src/main/java/eu/siacs/conversations/crypto/PgpDecryptionService.java#L106).

Under what circumstances would OpenKeychain return a pending intent asking the user to select a key as described above. In all testing with invalid data or encrypted content that doesn't include my key OpenKeychain would just return an error.

**Comment by dschuermann** [comment]
Unfortunately, I wasn't able to reproduce. The code looks okay, the API call with ``ACTION_DECRYPT_VERIFY`` should be enough. I am currently not using the OpenPGP encryption on a daily basis with Conversations, maybe that's why I haven't encountered this until now.

**Comment by Orhideous** [comment]
I can confirm this issue. Moreover, I'm unable to decrypt *my own* messages, mirrored from Gajim to Conversations via mod_carbon (prosody)

**Comment by akater** [comment]
Could be OpenKeychain issue?

Might be relevant: some keys are created by default with capabilities SC, and with separate S subkey, which makes 2 signing keys in total. (My key is like this.) Maybe key recognition might get messed up due to this. I'm not sure if that would make it an OpenKeychain issue, a Conversations issue, or an XMPP issue.

**Comment by Orhideous** [comment]
@akater I reproduced this bug without subkeys ([example](http://pgp.mit.edu/pks/lookup?op=vindex&search=0xC5234D755464EDA6) of such key)

**Comment by Orhideous** [comment]
Seems like I found why this issue happens.
When I have private key WITH password, first time OpenKeychain asks me for password and time span to remember it. Assume that I typed correct password and selected «remember until screen off». Then Conversations successfully decrypts all messages.
I lock my phone, screen turns off, and Conversations is unable to decrypt any subsequent PGP messages.

Workaround: kill app and wait when OpenKeychain asks for password again.

Sidenote: all messages are mirrored to my phone via mod_carbon.

@iNPUTmice, is it a Conversations bug or OpenKeychain bug?
